### PR TITLE
RPRBLND-1996: Improve Material X import/export

### PR DESCRIPTION
**Pull request by DagerD**
### PURPOSE
Improve MaterialX export by adding new options.

### EFFECT OF CHANGE
Added following options for MaterialX export:
1. Export MaterialX dependencies
2. Export bound textures
3. Folder name for bound textures

### TECHNICAL STEPS

- Operators modified with corresponded properties.
- separate export function to export_mx_to_file
- separate convert function to convert_mx_node_tree
- HDUSD_MATERIAL_OP_export_mx_file, HDUSD_MX_OP_export_file, HDUSD_MATERIAL_OP_convert_mx_node_tree now calls corresponded functions.

### CODE REVIEW

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Need code improvements

File: src/hdusd/ui/mx_nodes.py [hunk 1]
```diff
@@ -72,14 +72,34 @@ class HDUSD_MX_OP_export_file(HdUSD_Operator, ExportHelper):
...
+                                               description="Export bound textures to corresponded folder",
+                                               default=False)
+
+    texture_dir_name: bpy.props.StringProperty(
+        name="Texture folder name",
+        description="Texture folder name used for exporting files",
+        default='Textures',
+        maxlen=1024
+    )
+
```
[the remark this answers is not in the document]
Comment (DagerD) [reply src/hdusd/ui/mx_nodes.py:89]: no draw() here

**Review by VascoPi — APPROVED** [review]
Works well, some details were discussed offline.

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Export operator UI can be improved by using draw(). Other improvements.

**Review by bnagirniak — APPROVED** [review]
Tested - works good

**Review by bsavery — CHANGES_REQUESTED** [review]
Fix conflicts.

**Review by bnagirniak — APPROVED** [review]
Tested after conflicts fix - works ok

**Review by VascoPi — APPROVED** [review]
Works well.

**Pull request merged**